### PR TITLE
(#6586) move istio.io test for circuit breaking to istio.io repository

### PR DESCRIPTION
**Pull request by su225**
This PR includes istioio test for Circuit breaker
#6586

### CODE REVIEW

File: tests/trafficmanagement/circuitbreaker_test.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+					}),
+					SnippetInput: istioio.Evaluate(istioio.Path("scripts/trip_circuitbreaker.txt"), map[string]interface{}{
+						"isSnippet":                true,
+						"inputTerminalFlag":        "-it",
+						"beforeCircuitBreakVerify": "",
+						"afterCircuitBreakVerify":  "",
+						"outputRedirectionCmd":     "",
+					}),
+				}).
+			Defer(istioio.Script{Input: istioio.Path("scripts/circuitbreaker_test_cleanup.txt")}).
```
Comment (brian-avery) [inline tests/trafficmanagement/circuitbreaker_test.go]: I'd add verification of the istio-proxy stats to make sure they are in a similar format to what the user is looking for.

Comment (su225) [reply tests/trafficmanagement/circuitbreaker_test.go]: I see something similar in the actual tutorial.
```
$ kubectl exec $FORTIO_POD -c istio-proxy -- pilot-agent request GET stats | grep httpbin | grep pending
cluster.outbound|8000||httpbin.default.svc.cluster.local.circuit_breakers.default.rq_pending_open: 0
cluster.outbound|8000||httpbin.default.svc.cluster.local.circuit_breakers.high.rq_pending_open: 0
cluster.outbound|8000||httpbin.default.svc.cluster.local.upstream_rq_pending_active: 0
cluster.outbound|8000||httpbin.default.svc.cluster.local.upstream_rq_pending_failure_eject: 0
cluster.outbound|8000||httpbin.default.svc.cluster.local.upstream_rq_pending_overflow: 21
cluster.outbound|8000||httpbin.default.svc.cluster.local.upstream_rq_pending_total: 29
```
Is this what you meant by "istio-proxy stats" verification? For instance, verifying that count of `upstream_rq_pending_overflow` falls within some range besides parsing fortio output?

Comment (brian-avery) [reply tests/trafficmanagement/circuitbreaker_test.go]: Exactly :) The user is asked to run the command to look at the stats, so we just need to verify that the output looks similar.

**Comment by brian-avery** [comment]
This fixes https://github.com/istio/istio/pull/20099

**Comment by su225** [comment]
I noticed one more problem - the program `bc` is not available inside container and so the checks on percentage is failing. I'm exploring alternatives. `awk` is available. I'm gonna give that one a try.

File: tests/trafficmanagement/circuitbreaker_test.go (hunk 1 above)
Comment (su225) [reply tests/trafficmanagement/circuitbreaker_test.go]: I just added verification for output format. The counts look very erratic. When I was running tests, the percentage of requests getting 503 before tripping ranged from 70% to 100%. After tripping, it was even more erratic. Consequently `upstream_rq_pending_overflow` range was wide. Setting the range to be too wide defeats the purpose of it where as setting it to a narrow range would make the test flaky.

**Comment by su225** [comment]
Found out that the container has `ruby` installed. I used that for evaluating arithmetic expressions. Not sure if that is a good idea. Looks like using `eval`.

**Comment by su225** [comment]
@brian-avery - hey there, I'm getting linter errors repeatedly. Here is the interesting part in logs
```
level=warning msg="[runner] Can't run linter goanalysis_metalinter: assign: failed prerequisites: inspect@istio.io/istio.io/tests/trafficmanagement [istio.io/istio.io/tests/trafficmanagement.test]"
level=info msg="[runner] processing took 4.539µs with stages: max_same_issues: 926ns, skip_dirs: 514ns, nolint: 459ns, identifier_marker: 405ns, max_from_linter: 320ns, cgo: 254ns, filename_unadjuster: 249ns, path_prettifier: 230ns, skip_files: 168ns, autogenerated_exclude: 167ns, uniq_by_line: 151ns, diff: 144ns, exclude: 142ns, max_per_file_from_linter: 140ns, path_shortener: 129ns, exclude-rules: 77ns, source_code: 64ns"
level=info msg="[runner] linters took 31.743781129s with stages: goanalysis_metalinter: 31.743723834s"
level=error msg="Running error: assign: failed prerequisites: inspect@istio.io/istio.io/tests/trafficmanagement [istio.io/istio.io/tests/trafficmanagement.test]"
```
First time, it is `interfacer` and this time it is `insepct`.

Digging in, I found that there could be some issues related to golangci. Here is the issue
https://github.com/golangci/golangci-lint/issues/840

**Comment by su225** [comment]
@brian-avery - Linting is failing because of some broken link in Brazilian Portuguese documentation
```
- ./public/pt-br/faq/applications/index.html
  *  External link https://zookeeper.apache.org/doc/r3.5.7/zookeeperAdmin.html failed: 404 No error
```
Rest of the things are fine. So ok for merge?

**Pull request merged**